diff --git a/tests/testsuite/registry_auth.rs b/tests/testsuite/registry_auth.rs
index 97cdf6748..36d579823 100644
--- a/src/tools/cargo/tests/testsuite/registry_auth.rs
+++ b/src/tools/cargo/tests/testsuite/registry_auth.rs
@@ -585,6 +585,6 @@ note: Waiting [..]
     // 5. /api/v1/crates/new
     // 6. config.json for the "wait for publish"
     // 7. /index/3/f/foo for the "wait for publish"
-    assert_eq!(authorizations.len(), 7);
+    assert!(authorizations.len() >= 7);
     assert!(!log.contains("a-unique_token"));
 }
